File: mobile/lib/model/offer.dart
import 'package:client/model/property.dart';
import 'package:client/model/user.dart';
import 'package:json_annotation/json_annotation.dart';
part "offer.g.dart";

@JsonSerializable()
class Offer {
  int id;
  int landlordId;
  int customerId;
  int? propertryId;
  int userCreatedId;
  DateTime offerDate;
  DateTime offerExpires;
  String description;
  int price;
  String offerStatus;
  User? landlord;
  User? customer;
  User? userCreated;
  Property? property;

  Offer({
    required this.id,
    required this.landlordId,
    required this.customerId,
    required this.propertryId,
    required this.userCreatedId,
    required this.offerDate,
    required this.offerExpires,
    required this.description,
    required this.offerStatus,
    required this.price,
    this.landlord,
    this.customer,
    this.property,
    this.userCreated,
  });

  factory Offer.fromJson(Map<String, dynamic> json) => _$OfferFromJson(json);
  Map<String, dynamic> toJson() => _$OfferToJson(this);
}

File: server/Models/Offer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using RedHouse_Server.Models;

namespace server.Models
{
    public class Offer
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(Landlord))]
        public int LandlordId { get; set; }
        public User Landlord { get; set; }

        [ForeignKey(nameof(Customer))]
        public int CustomerId { get; set; }
        public User Customer { get; set; }
        
        [ForeignKey(nameof(Property))]
        public int PropertyId { get; set; }
        public Property Property { get; set; }
        public int UserCreatedId { get; set; }
        public DateTime OfferDate { get; set; }
        public DateTime OfferExpires { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string OfferStatus { get; set; }
    }

}
